### ISSUE TITLE
Decision to be taken : Individual decision system

### ISSUE THREAD

**Issue by samuelcardillo** [RTFKT-Ltd/anti-theft#3]
**This is a major point of discussion** 

Allowing individual decisions would make the contract more flexible. Allowing collectors to do things like, potentially : 

- Add a whitelist address to bypass the timelock
- Activate/deactivate the anti-theft system for themselves
- etc...

The main issue comes from the fact that, if the wallet itself is compromised (aka the passphrase was stolen, and the hack is just not coming from a `setApprovalForAll`), then the attacker can play with those individual parameters, rendering the anti-theft system only good against the `setApprovalForAll` attacks.

**Comment by krjii** [comment]
I think the scenario where passphrase is compromised is a different scenario in and of itself. To address the various aspects of security, the discussion should be broken down into tiers of security. So the tiers would be something like wallet security, collection security, and exchange security. Or is there a better way to categorize the security concerns?

**Comment by mongodillo** [comment]
> I think the scenario where passphrase is compromised is a different scenario in and of itself. To address the various aspects of security, the discussion should be broken down into tiers of security. So the tiers would be something like wallet security, collection security, and exchange security. Or is there a better way to categorize the security concerns?

I agree that we should classify these wallet security compromises into different buckets with different solutions, namely the `setapprovalforall/signing txns/msgs` and stolen seed phrase buckets.

1. Setapprovalforall/signing transactions/msgs - the main reason people get scammed this way is because they do not know what they are signing (be it in messages or txns). 

The recent Opensea phishing as well as ongoing scams we see all take advantage of this fact. 

Perhaps a solution here is to have some form of browser extension or middle layer which translates the messages into simple layman terms. The current eip712 structure is still very complex for most and I believe majority of people still have no idea what they are signing. 

2. Seed phrase compromised - this bucket of issues can be due to keyloggers, malware, social engineering etc. 


The whitelisting of addresses seems like a good idea to tackle this front as to who the asset can be transferred to, but perhaps any user should have a separate "recovery code" in the anti theft contract to authorize the addition of new addresses too?

**Comment by krjii** [comment]
Yep, you're thinking like I'm thinking. I'll take some time these week to put a little more thought into it. May e we can mastermind an implementation. I was thinking a browser extension as well to help translate what users are signing. Ideally a tool that could support cold and hot storage.
